### ISSUE TITLE
setComplete method triggers 'TTL caching without ttlAutopurge, max, or maxSize' error with no available options to control it

### ISSUE THREAD

**Issue by Jangyusu** [prometheus/prometheus#15396]
### What did you do?

I'm encountering an issue with the setComplete method when using PromQLExtension

### What did you see instead? Under which circumstances?

The method triggers the following error: TTL caching without ttlAutopurge, max, or maxSize can result in unbounded memory consumption.

After reviewing the documentation and source code, it seems there are no available options to configure or limit the cache behavior. This could potentially lead to significant memory issues in long-running applications.

It would be helpful to provide configurable options such as ttlAutopurge, max, or maxSize to mitigate this risk. Please consider adding this functionality.

**Comment by bboreham** [comment]
Hello from the bug-scrub!

@juliusv @Nexucis since you know most about the UI, could you comment on this please?

**Comment by Nexucis** [comment]
@Jangyusu can you give more information about what config did you pass to the method `setComplete` ?

I would believe somehow the cache config is not well set and that would trigger this error or warning (that's at least what the TTL documentation is saying, it would raise a warning not an error).

There is no such error or warning in the Prometheus UI currently. So it should be something you are doing or not setting in the config.

Please provide more context so we can provide meaningful advices.

**Comment by juliusv** [comment]
@Nexucis You can see this error in the browser console on any Prometheus server:

<img width="1910" height="569" alt="Image" src="https://github.com/user-attachments/assets/9bee47ee-9a8b-47ac-aab2-2395bb9c02b5" />

I've seen this there for a long time, but since it just seems to be a warning, I've just ignored it until now.

**Comment by Nexucis** [comment]
This is really strange, I can't see it on my side...
Well, I guess I will have to investigate more

**Comment by Nexucis** [comment]
I just try again on https://prometheus.demo.prometheus.io and I don't see the error in the console.

I am using Google Chrome Version 142.0.7444.60 (Official Build) (arm64)

@juliusv do you still see it ? And if yes which browser are you using ?

**Comment by juliusv** [comment]
@Nexucis Indeed, I get it with older versions like 3.2, but not on current main anymore. So I guess this can be closed.
